### ISSUE TITLE
Deno runtime not supported

### ISSUE THREAD

**Issue by ulken** [75lb/command-line-usage#40]
Due to https://github.com/denoland/deno/issues/13321, the library fails with

```sh
error: Uncaught TypeError: chalk.underline.bold is not a function
    at ContentSection.header (file:///Users/ulken/Library/Caches/deno/npm/registry.npmjs.org/command-line-usage/6.1.3/lib/section.js:23:32)
```

The version of `chalk` used is really old. Please consider upgrading to at least 4.X, where the issue above is fixed. If you want to take the step to ESM, you might want to take it to 5.X+ even.

If you're up for it, let me know if you want me to help out in any way.

PS. Even if you have no intention of supporting Deno, I think it would still be worth to upgrade Chalk to a more modern (and safer) version.

**Comment by 75lb** [comment]
Hi, could you send some simple instructions on how to reproduce this please..

**Comment by ulken** [comment]
> 

Sure thing. Would you prefer a sandbox or instructions how to reproduce it locally?

**Comment by 75lb** [comment]
instructions on how to reproduce it locally please..

* What pre-reqs do I need to install?
* Post some code which fails
* Post a command I can run to see it failing and then later run again to check the fix works..

**Comment by ulken** [comment]
Got it! I'll get back to you.

**Comment by ulken** [comment]
1. [Install Deno](https://deno.land/manual@v1.30.3/getting_started/installation)
2. `mkdir cli-usage-deno && cd $_`
3. `deno init`
4. Replace contents of `main.ts` with
```ts
import cliUsage from "npm:command-line-usage";

if (import.meta.main) {
  console.log(cliUsage([
    { 
      header: "Deno CLI usage", 
      content: "Best thing ever"
    }
  ]));
}
```
5. `deno run --allow-read --allow-env main.ts` (could also pass `--allow-all` if you prefer)

Let me know if you get stuck on something.

**Comment by ulken** [comment]
Let me know if a PR would help.

**Comment by 75lb** [comment]
I had a look last night, chalk v5 is ESM only, meaning the command-line-usage project would need to be upgraded to a ESM project (`"type": "module"`).. which is fine, but it would also still need to support CJS as we don't have an "ESM only" policy on these projects meaning rollup would need to be introduced to build a CJS option, which is fine too but it's probably not as quick a fix as you were hoping

**Comment by ulken** [comment]
Right, which is why I wrote:

> Please consider upgrading to at least 4.X, where the issue above is fixed. If you want to take the step to ESM, you might want to take it to 5.X+ even.

From the Chalk README:

> IMPORTANT: Chalk 5 is ESM. If you want to use Chalk with TypeScript or a build tool, you will probably want to use Chalk 4 for now.

and

> It's totally fine to stay on Chalk v4. It's been stable for years.

I.e v4 should solve the problem equally well.

**Comment by ulken** [comment]
> also, the tagged template support [was removed](https://github.com/chalk/chalk/releases/tag/v5.0.0) from chalk v5.. this library uses that feature to give users the ability to colour usage docs, e.g. `some {red important} text`.. so a bit of extra work required there too.. 

Moved to a separate package, though, so shouldn't be too bad?

**Comment by 75lb** [comment]
i thought the Deno fix was from [chalk v5.2 onward](https://github.com/chalk/chalk/releases/tag/v5.2.0).. may be wrong..

**Comment by 75lb** [comment]
since implementing this fix will introduce a breaking change to command-line-usage, we might as upgrade all deps to all latest versions (v5, not v4 chalk) and rewrite the project as ESM.. that way we can squeeze as many breaking upgrades into the next release as possible.. since this update will result in changes right across the project (in /examples and /test too) I will do it this time.. otherwise, yes I'd welcome your PR..

**Comment by ulken** [comment]
All right, by all means. Thank you for taking the leap. You know where to find me. Good luck!

**Comment by 75lb** [comment]
I've published a beta version for testing.. please update your `import` with the following and give it a try.. 

```
import cliUsage from "npm:command-line-usage@next"
```

**Comment by ulken** [comment]
Hey, look at that!

<img width="317" alt="Screenshot 2023-02-14 at 16 50 53" src="https://user-images.githubusercontent.com/516549/218788804-c04bb7b8-a2ae-4049-830c-abae5150a226.png">

For unknown reason, the formatting is not displayed in Warp, but that is on them (works in Hyper and Terminal.app)... 

https://app.warp.dev/block/nvpi0MT2iXS45OmFaVQyjT

Is it erroneously thinking it's piped and strips the formatting? Idk. Don't bother.

**Comment by 75lb** [comment]
I'm about to move this out of beta and make it live, please let me know if there are any further deno issues I might not know about.. Cheers!

**Comment by 75lb** [comment]
Released in [v7.0.0](https://github.com/75lb/command-line-usage/releases/tag/v7.0.0)

**Comment by ulken** [comment]
Cool! Will do 👍
